src/mainwindow.cpp

```diff
@@ -151,7 +152,7 @@ int MainWindow::getImageRotation()
 
 void MainWindow::updateImage(bool immediately)
 {
```

Comment:
couldn't we draw the image here and immediately render the next one in the same thread? That way we just had to make sure we don't run updateImage() again before the previous run is finished (when the render took longer than the timeout)

Reply:
I think that would definitely be better. Tried this idea previously but could not get it to work, reason being (it seems to me) `QWidget::update()` will not draw the image until the function `ImageSwitcher::updateImage()` returns. 
Any suggestions or it's just me not familiar with QT.
Also tried offloading the rendering to a third thread, but that adds more coordination and I can open up another PR for it if desire.